qfieldsync/gui/cloud_transfer_dialog.py

```diff
@@ -538,7 +569,12 @@ def build_files_tree(self):
                 # the length of the stack and the parts is equal for file entries
                 if len(stack) == len(parts):
                     item.setData(0, Qt.UserRole, project_file)
-                    self.add_file_checkbox_buttons(item, project_file)
+                    self.add_file_checkbox_buttons(
+                        item,
+                        project_file,
+                        project_file.local_path_exists
+                        and str(project_file.local_path) in offline_layers_paths,
+                    )
```

Comment:
```suggestion
                    is_offline_layer = (
                        project_file.local_path_exists
                        and str(project_file.local_path) in offline_layers_paths
                    )

                    self.add_file_checkbox_buttons(item, project_file, is_offline_layer)
```

Took me a while to count the number of paratemers as 3 instead of 4, so I guess a good candidate to calculate the flag in a separate step.

Reply:
Didn't look too much in details, but I would assume we cannot add `LayerSource.is_offline_layer` flag that moves the logic where it makes most sense I guess?